extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -84,17 +92,48 @@ public void setConfig(final TransactionManagerConfiguration transactions) {
                 .setExpiryScannerClassNames(transactions.expiryScanners);
         BeanPopulator.getDefaultInstance(JTAEnvironmentBean.class)
                 .setXaResourceOrphanFilterClassNames(transactions.xaResourceOrphanFilters);
+        if (transactions.objectStoreType.equals(ObjectStoreType.JDBC)) {
+            String dataSource = pickDataSource(transactions, jdbcDataSources);
+            Arc.container().instance(DataSource.class, NamedLiteral.of(dataSource)).get();
```

Comment:
That's good for named datasources.

Careful though, I don't think the default datasource is named, at least the CDI bean isn't. See https://github.com/quarkusio/quarkus/blob/d0ea5ca5dd6c3bd94344fede4bf092d1fdd78830/extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java#L253-L272

To retrieve the default datasource you'd need to use:

```
Arc.container().instance(DataSource.class);
```

Reply:
I use `QuarkusDataSource` as a wrapper which could be lazy initialized.  And handle the defautl datasource as you suggested. Thanks!